cmd/serve.go

```diff
@@ -0,0 +1,326 @@
+	Run: func(cmd *cobra.Command, args []string) {
+		// Load environment configuration
+		envConfig, err := config.LoadEnvConfig(config.GetEnvPath())
+		if err != nil {
+			log.Fatalf("Error loading environment configuration: %v", err)
+		}
+
+		// Get server configuration
+		serverConfig := envConfig.GetServerConfig()
+		if serverConfig == nil {
+			log.Fatal("Server configuration not found. Please run 'comanda configure --server' first")
+		}
+
+		// Create data directory if it doesn't exist
+		if err := os.MkdirAll(serverConfig.DataDir, 0755); err != nil {
+			log.Fatalf("Error creating data directory: %v", err)
+		}
+
+		// Health check endpoint
+		http.HandleFunc("/health", logRequest(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Type", "application/json")
+			json.NewEncoder(w).Encode(HealthResponse{
+				Status:    "ok",
+				Timestamp: time.Now().Format(time.RFC3339),
+			})
+		}))
+
+		// List files endpoint
+		http.HandleFunc("/list", logRequest(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Type", "application/json")
+
+			if !checkAuth(serverConfig, w, r) {
+				return
+			}
+
+			files, err := filepath.Glob(filepath.Join(serverConfig.DataDir, "*.yaml"))
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+				json.NewEncoder(w).Encode(ListResponse{
+					Success: false,
+					Error:   fmt.Sprintf("Error listing files: %v", err),
+				})
+				return
+			}
+
+			// Convert absolute paths to relative paths
+			var relFiles []string
+			for _, file := range files {
+				relFile, err := filepath.Rel(serverConfig.DataDir, file)
+				if err != nil {
+					continue
+				}
+				relFiles = append(relFiles, relFile)
+			}
+
+			json.NewEncoder(w).Encode(ListResponse{
+				Success: true,
+				Files:   relFiles,
+			})
+		}))
+
+		// Process endpoint
+		http.HandleFunc("/process", logRequest(func(w http.ResponseWriter, r *http.Request) {
+			if !checkAuth(serverConfig, w, r) {
+				return
+			}
+			handleProcess(w, r, serverConfig)
+		}))
+
+		fmt.Printf("Starting server on port %d...\n", serverConfig.Port)
+		fmt.Printf("Data directory: %s\n", serverConfig.DataDir)
+		if serverConfig.Enabled {
+			fmt.Println("Authentication is enabled. Bearer token required.")
+			fmt.Printf("Example usage: curl -H 'Authorization: Bearer %s' 'http://localhost:%d/process?filename=examples/openai-example.yaml'\n",
+				serverConfig.BearerToken, serverConfig.Port)
+		} else {
+			fmt.Printf("Example usage: curl 'http://localhost:%d/process?filename=examples/openai-example.yaml'\n", serverConfig.Port)
+		}
+
+		if err := http.ListenAndServe(fmt.Sprintf(":%d", serverConfig.Port), nil); err != nil {
+			log.Fatalf("Server failed to start: %v", err)
+		}
+	},
+}
+
+func handleProcess(w http.ResponseWriter, r *http.Request, serverConfig *config.ServerConfig) {
+	w.Header().Set("Content-Type", "application/json")
+
+	// Only allow GET requests
+	if r.Method != http.MethodGet {
```

Comment:
I think whether this is a state changing operation depends what's in the action tag of the YAML file - this is where RESTful design doesn't exactly mesh with the flexibility of prompted actions. I'm going to leave it as GET for now because most use cases are *likely* about synthesizing data to return information.